model/jpa/src/main/java/org/keycloak/models/jpa/session/JpaUserSessionPersisterProvider.java

```diff
@@ -240,26 +254,19 @@ public void updateAllTimestamps(int time) {
         }
 
         List<PersistentUserSessionEntity> results = query.getResultList();
-        List<UserSessionModel> result = new ArrayList<>();
+
+        List<PersistentUserSessionAdapter> result = new ArrayList<>();
```

Comment:
+1. I think it will require to add new method "toAdapter" with PersistentUserSessionEntity and realm will need to be lookup from that method, but that should be fine. Will try to update it this way.